felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -461,6 +476,43 @@ func (m *bpfEndpointManager) onInterfaceAddrsUpdate(update *ifaceAddrsUpdate) {
 	}
 }
 
+func (m *bpfEndpointManager) wepIfaceUpdateIndex(iface string, ifindex int) {
+	oldIdx, ok := m.ifaceNameToIdx[iface]
+	if !ok || oldIdx.ifindex != ifindex {
+		val := ifaceIdx{
+			ifindex:  ifindex,
+			workload: m.isWorkloadIface(iface),
+		}
+		m.ifaceNameToIdx[iface] = val
+		flags := uint32(0)
+		if val.workload {
+			flags = ifstate.FlgWEP
+		}
+
+		kBytes := ifstate.NewKey(uint32(ifindex)).AsBytes()
+
+		// If we get an update and the wep is alredy marked as ready.
+		oldValBytes := m.ifStateMap.GetDesired(kBytes)
+		if oldValBytes != nil {
+			var oldVal ifstate.Value
+			copy(oldVal[:], oldValBytes)
+			if oldName := oldVal.IfName(); oldName != iface {
+				log.Debugf("dev %s reuses ifindex %d previously %s.", iface, ifindex, oldName)
+			} else if oldVal.Flags()&ifstate.FlgReady != 0 {
+				flags |= ifstate.FlgReady
```

Comment:
I think this might all be a bit simpler if you stored the index and readiness flag in `m.nameToIface`, then you wouldn't need to parse back the entry and merge the flags, you could just do a blind `SetDesired` and let the map worry about deltas.